### ISSUE TITLE
Duration changes with FPS

### ISSUE THREAD

**Issue by Meldiron** [mifi/editly#49]
Using JSON below I tried many different renders to see how framerate changes video duration. I will try to find bug in code but I may need some help. Here are some tests:

## Desired duration: 90 seconds

| width (px) | height(px) | fps | output duration (sec) | Amount of tests |
|------------|------------|-----|-----------------------|-----------------|
| 400        | 400        | 10  | 88.700000             | 2               |
| 400        | 400        | 20  | 88.850000             | 2               |
| 400        | 400        | 30  | 88.900000             | 1               |
| 400        | 400        | 40  | 88.925000             | 1               |
| 400        | 400        | 50  | 88.940000             | 3               |
| 400        | 400        | 60  | 88.950000             | 1               |
| 400        | 400        | 120 | 88.975000             | 1               |
| 400        | 400        | 240 | 88.988000             | 1               |
|------------|------------|-----|-----------------------|-----------------|
| 800        | 800        | 10  | 88.700000             | 1               |
| 1920       | 1080       | 10  | 88.700000             | 1               |

JSON:
```
{ 
  "width": 1920, "height": 1080, "fps": 10,
  "outPath": "./subtitle.mp4",
  "defaults": {
    "layer": { "fontPath": "./Montserrat-Bold.ttf" },
    "transition": { "name": "fade" },
  },
  "clips": [
    { "duration": 30, "layers": [
       {
          "type": "subtitle",
          "text": "ST1 "
        },
         {
          "type": "title",
          "text": "T1 "
        }
    ] },
    { "duration": 30, "layers": [
       {
          "type": "subtitle",
          "text": "ST2 "
        },
         {
          "type": "title",
          "text": "T2 "
        }
    ] },
    { "duration": 30, "layers": [
       {
          "type": "subtitle",
          "text": "ST3 "
        },
         {
          "type": "title",
          "text": "T3 "
        }
    ] },
  ],
}
```

**Comment by mifi** [comment]
Yeah, I think there is a small bug where it drops a frame some times. Will look into it when I have more time

**Comment by swlemp** [comment]
@mifi Can confirm when using `framerate: 60`

**Comment by TimoGruber** [comment]
Can also confirm when using fps: 15 and fps: 30 on my videos -> expected duration 18:17 -> output duration 18:15

**Comment by swlemp** [comment]
@mifi I have two videos, both are 1.5 seconds, 60 fps.  I do a 3-second transition so it should be as flush 3-sec output.  Instead, the output is 2.21667s.  When I change the FPS to 15, the output becomes 2.2s.

It's causing my video and audio to be out of sync in proportion to the number of transitions.  I tried commenting out the pts factor and scaling stuff in `videoFrameSource.js` and it made no difference though.

**Comment by swlemp** [comment]
@TimoGruber For now I am using ffmpeg-gl-transitions since gl transitions were the only component of editly I was using

**Comment by mifi** [comment]
Can you try again in 0.5.0?

**Comment by TimoGruber** [comment]
For me there is still a slightly difference on the output file (about 1 sec). But with the ```keepSourceAudio``` option there is obviously no need for creating a external audio file from the sources and sync it with the video.
